### PR TITLE
Make css.Barrier._has_unitary_() return False

### LINKED ISSUE #314: Make css.Barrier._has_unitary() return False
Moving this https://github.com/SupertechLabs/general-superstaq/issues/118 issue into cirq-superstaq

### PR DESCRIPTION
**Pull request by SalahedeenIssa**
Fixes #314

### CODE REVIEW

File: cirq_superstaq/custom_gates.py [hunk 1]
```diff
@@ -91,6 +91,7 @@ def _resolve_parameters_(self, resolver: cirq.ParamResolver, recursive: bool) ->
         )
 
     def _has_unitary_(self) -> bool:
+        print('yes')
```
Comment (vtomole) [inline cirq_superstaq/custom_gates.py:94]: Remove debug print

**Review by vtomole — CHANGES_REQUESTED** [review]
Add test

File: cirq_superstaq/custom_gates_test.py [hunk 2]
```diff
@@ -312,6 +312,7 @@ def test_barrier() -> None:
 
     barrier = css.barrier(*qubits)
     assert barrier == css.Barrier(n).on(*qubits)
+    assert not gate._has_unitary_()
```
Comment (vtomole) [inline cirq_superstaq/custom_gates_test.py:315]: Do `assert not cirq.has_unitary(gate)`

**Review by vtomole — DISMISSED** [review]
LGTM

**Comment by perlinm** [comment]
This possibly introduces more problems than it solves.  Tentatively closing and filing [ss#1520](https://github.com/SupertechLabs/SuperstaQ/issues/1520)

**Pull request closed without merging**